espnet/nets/pytorch_backend/frontend_asr.py

```diff
@@ -0,0 +1,212 @@
+import argparse
+from typing import Tuple
+from typing import Union
+
+import configargparse
+import numpy as np
+import torch
+
+from espnet.nets.asr_interface import ASRInterface
+from espnet.nets.asr_interface import FrontendASRInterface
+from espnet.nets.pytorch_backend.frontends.feature_transform import FeatureTransform
+from espnet.nets.pytorch_backend.frontends.frontend import Frontend
+from espnet.nets.pytorch_backend.nets_utils import pad_list
+from espnet.nets.pytorch_backend.nets_utils import to_device
+from espnet.nets.pytorch_backend.nets_utils import to_torch_tensor
+from espnet.utils.cli_utils import strtobool
+
+
+class FrontendASR(FrontendASRInterface, torch.nn.Module):
+    """An implementation of FrontendASR with WPE and MVDR"""
+    @staticmethod
+    def add_arguments(parser: Union[argparse.ArgumentParser,
+                                    configargparse.ArgumentParser]):
+        # WPE related
+        parser.add_argument('--use-wpe', type=strtobool, default=False,
+                            help='Apply Weighted Prediction Error')
+        parser.add_argument('--wtype', default='blstmp', type=str,
+                            choices=['lstm', 'blstm', 'lstmp', 'blstmp', 'vgglstmp', 'vggblstmp', 'vgglstm', 'vggblstm',
+                                     'gru', 'bgru', 'grup', 'bgrup', 'vgggrup', 'vggbgrup', 'vgggru', 'vggbgru'],
+                            help='Type of encoder network architecture '
+                                 'of the mask estimator for WPE. '
+                                 '')
+        parser.add_argument('--wlayers', type=int, default=2,
+                            help='')
+        parser.add_argument('--wunits', type=int, default=300,
+                            help='')
+        parser.add_argument('--wprojs', type=int, default=300,
+                            help='')
+        parser.add_argument('--wdropout-rate', type=float, default=0.0,
+                            help='')
+        parser.add_argument('--wpe-taps', type=int, default=5,
+                            help='')
+        parser.add_argument('--wpe-delay', type=int, default=3,
+                            help='')
+        parser.add_argument('--use-dnn-mask-for-wpe', type=strtobool,
+                            default=False,
+                            help='Use DNN to estimate the power spectrogram. '
+                                 'This option is experimental.')
+        # Beamformer related
+        parser.add_argument('--use-beamformer', type=strtobool,
+                            default=True, help='')
+        parser.add_argument('--btype', default='blstmp', type=str,
```

Comment:
Can you make these `--bxxx` more expressive like `--masknet-type` or something? because we have already added breaking change in the option `--use-frontend`